tsk/fs/logical_fs.cpp

```diff
@@ -274,19 +274,57 @@ populate_fs_file_from_win_find_data(const WIN32_FIND_DATA* fd, TSK_FS_FILE * a_f
 TSK_TCHAR * create_search_path(const TSK_TCHAR *base_path) {
 	size_t len = TSTRLEN(base_path);
 	TSK_TCHAR * searchPath;
-	searchPath = (TSK_TCHAR *)tsk_malloc(sizeof(TSK_TCHAR) * (len + 4));
+	size_t searchPathLen = len + 4;
+	searchPath = (TSK_TCHAR *)tsk_malloc(sizeof(TSK_TCHAR) * (searchPathLen));
 	if (searchPath == NULL) {
 		return NULL;
 	}
-	TSTRNCPY(searchPath, base_path, len + 1);
+
 #ifdef TSK_WIN32
-	TSTRNCAT(searchPath, L"/*", 3);
+	TSTRNCPY(searchPath, base_path, len + 1);
+	TSTRNCAT(searchPath, L"\\*", 4);
 #else
+	TSTRNCPY(searchPath, base_path, len + 1);
 	TSTRNCAT(searchPath, "/*", 3);
 #endif
 	return searchPath;
 }
 
+/*
+* Create the wildcard search path used to find directory contents using
+* the absolute directory and unicode prefix. We only call this method for
+* long paths because it does not work in cygwin - prepending "\\?\" only
+* works for absolute paths starting with a drive letter.
+*
+* @param base_path The path to the directory to open
+*
+* @return The search path with wildcard appended (must be freed by caller)
+*/
+TSK_TCHAR * create_search_path_long_path(const TSK_TCHAR *base_path) {
+#ifdef TSK_WIN32
+
+	// First convert the base path to an absolute path
+	TCHAR absPath[LOGICAL_MAX_PATH_UNICODE];
```

Comment:
do we need to add +1 here for the null byte?

Reply:
I don't believe so? https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getfullpathnamew